test/spanner.ts

```diff
@@ -2890,6 +2927,30 @@ describe('Spanner with mock server', () => {
       });
     });
 
+    it('should retry on aborted when running parallel query', async () => {
+      let attempts = 0;
+      const database = newTestDatabase();
+      const rowCount = await database.runTransactionAsync(
+        (transaction): Promise<number> => {
+          if (!attempts) {
+            spannerMock.abortTransaction(transaction);
+          }
+          attempts++;
+          return Promise.all([
+            transaction!.run(selectSql),
+            transaction!.run(selectSql),
+          ]).then(([rows]) => {
+            let count = 0;
+            rows.forEach(() => count++);
+            return transaction.commit().then(() => count);
+          });
+        }
+      );
+      assert.strictEqual(rowCount, 3);
```

Comment:
Why is the row count 3? Shouldn't that be 2, as each query returns 1 row.

Reply:
Updated